HousingRepairsSchedulingApi/Gateways/DrsAppointmentGateway.cs

```diff
@@ -0,0 +1,57 @@
+namespace HousingRepairsSchedulingApi.Gateways
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Ardalis.GuardClauses;
+    using Domain;
+    using Domain.Drs;
+    using Services.Drs;
+
+    public class DrsAppointmentGateway : IAppointmentsGateway
+    {
+        private readonly int requiredNumberOfAppointmentDays;
+        private readonly int appointmentSearchTimeSpanInDays;
+        private readonly int appointmentLeadTimeInDays;
+        private readonly IDrsService drsService;
+
+        public DrsAppointmentGateway(IDrsService drsService, int requiredNumberOfAppointmentDays, int appointmentSearchTimeSpanInDays, int appointmentLeadTimeInDays)
+        {
+            Guard.Against.Null(drsService, nameof(drsService));
+            Guard.Against.NegativeOrZero(requiredNumberOfAppointmentDays, nameof(requiredNumberOfAppointmentDays));
+            Guard.Against.NegativeOrZero(appointmentSearchTimeSpanInDays, nameof(appointmentSearchTimeSpanInDays));
+            Guard.Against.Negative(appointmentLeadTimeInDays, nameof(appointmentLeadTimeInDays));
+
+            this.drsService = drsService;
+            this.requiredNumberOfAppointmentDays = requiredNumberOfAppointmentDays;
+            this.appointmentSearchTimeSpanInDays = appointmentSearchTimeSpanInDays;
+            this.appointmentLeadTimeInDays = appointmentLeadTimeInDays;
+        }
+
+        public async Task<IEnumerable<AppointmentSlot>> GetAvailableAppointments(string sorCode, string locationId,
```

Comment:
Is it appropriate to make this return a HACT object or do we want to decouple the domain of the application from the HACT data standard, and if so why?

Reply:
I wanted to keep the gateway decoupled from the domain to create another layer of abstraction between HACT, application and 3rd party service (DRS).

Once implemented, I was unsure if it was necessary.

Happy to remove `AppointmentSlot` and restore to using the HACT standard throughout and having the gateway convert the service response to HACT.

Reply:
As discussed, removed extra, unnecessary domain layer.